lib/Dialect/FIRRTL/FIRRTLOps.cpp

```diff
@@ -363,6 +358,18 @@ LogicalResult CircuitOp::verify() {
             << "previous extmodule definition occurred here";
         return failure();
       }
+      if (!aType.isa<FIRRTLBaseType>() || !bType.isa<FIRRTLBaseType>())
+        return extModule.emitOpError().append(
+                   "with 'defname' attribute ", defname,
+                   " has a port that is of unsupported type, must be base "
+                   "type."),
+               failure();
```

Comment:
In order to do this, needed to set the lambda type explicitly.  Reworked diagnostics in this lambda to take advantage of this, instead of split across few statements.

Reply:
Oh yeah, didn't realize we were in a lambda. I think it will be way nicer this way, so thanks for making the change!